### ISSUE TITLE
[3.3.5a] Ormorok the Tree Shaper Crystal Spikes always in Same Direction

### ISSUE THREAD

**Issue by wotlk-enthusiast** [TrinityCore/TrinityCore#24370]
**Description:**

When fighting the boss Ormorok the Tree Shaper in The Nexus, the crystal spikes are always in the same direction from the boss (generally away from the platform)

**Current behaviour:**

The crystal spikes radiate outward away from the approach platform.  This means that the entire party can just stay in one direction of the boss and miss the encounter mechanic entirely.

**Expected behaviour:**

The crystal spikes should radiate outward from the boss towards the players, forcing them to move.

**Steps to reproduce the problem:**

1. Walk into the nexus
2. Engage Ormorok
3. Don't kill him right away, and pay attention to which way the spikes go. 
4. Reposition him, either position or orientation, and see that the spikes always go the same direction from the boss.

**Branch(es):**

3.3.5a

**TC rev. hash/commit:** 

TrinityCore rev. b166d12 2020-03-21 21:14:49 +0100 (3.3.5 branch) (Unix, RelWithDebInfo, Static)
Online players: 0 (max: 6)
Active connections: 0 (max: 6) Queued connections: 0 (max: 0)
Server uptime: 1 Day 10 Hours 22 Minutes 59 Seconds.
Update time diff: 50.
**Operating system:** Linux

**Comment by Rushor** [comment]
is this not intended?

**Comment by wotlk-enthusiast** [comment]
No, the spikes are supposed to go out in different directions.  Here's a video of the encounter: https://www.youtube.com/watch?v=lU_2AoqlvhI

**Comment by wotlk-enthusiast** [comment]
So, I'd love to fix this, but am unsure where to start.  I sorta understand the code in [src/server/scripts/Northrend/Nexus/Nexus/boss_ormorok.cpp](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/Nexus/Nexus/boss_ormorok.cpp), but I'm not sure how it chooses the direction for the spikes. I'm also not sure a lowly unproven bug-opener like myself should be modifying the cpp without knowing what I'm doing (aka making more work for someone else)

**Comment by illfated** [comment]
There is also some old SQL content related to this:
https://github.com/TrinityCore/TrinityCore/blob/60a8b27226af0e186a62fb9d5116e4af98152378/sql/old/3.3.5a/TDB49_to_TDB50_updates/world/2012_11_18_00_world_ormorok.sql#L1-L56
(I am not sure if or how much this content affects the issue, but I thought I should mention it, just in case.)

**Comment by wotlk-enthusiast** [comment]
Just checked all that SQL. Where you have 
```sql
-- Apply aura to triggers 
 DELETE FROM `creature_template_addon` WHERE `entry` IN (27101,27079); 
 INSERT INTO `creature_template_addon`(`entry`,`auras`) VALUES 
 (27101,'47941'), -- Aura to summon the next trigger after 250ms 
 (27079,'47941'); -- Aura to summon the next trigger after 250ms and visual effect 
```

I have
```
+-------+-------+
| entry | auras |
+-------+-------+
| 27079 |       |
| 27101 |       |
+-------+-------+
```
I don't think that will affect the spell direction whatsoever.

With that noted exception, everything else is already appears in the WDB, with the comments field in smart_scripts being a little different.

**Comment by wotlk-enthusiast** [comment]
Can anyone tell me what this line does? 
https://github.com/TrinityCore/TrinityCore/blob/f1e4ee21f8da41c71f4cbafaa85e89c6275c1db2/src/server/scripts/Northrend/Nexus/Nexus/boss_ormorok.cpp#L283
